source/common/router/rds_impl.cc

```diff
@@ -249,6 +296,19 @@ void RdsRouteConfigProviderImpl::validateConfig(
   ConfigImpl validation_config(config, factory_context_, validator_, false);
 }
 
+// Schedules a VHDS request on the main thread and queues up the callback to use when the VHDS
+// response has been propagated to the worker thread that was the request origin.
+void RdsRouteConfigProviderImpl::requestVirtualHostsUpdate(
+    const std::string& for_domain,
+    Http::RouteConfigUpdatedCallbackSharedPtr route_config_updated_cb) {
+  auto alias =
+      VhdsSubscription::domainNameToAlias(config_update_info_->routeConfigName(), for_domain);
+  factory_context_.dispatcher().post(
+      [this, alias]() -> void { subscription_->updateOnDemand(alias); });
+  config_update_callbacks_->getTyped<ThreadLocalCallbacks>().callbacks_.push_back(
```

Comment:
Yeah, I'm sure we discussed this at some point, but would like to ask to be reminded about why we do TLS here vs. posting to/from the main thread and request thread dispatchers.

Reply:
I don't think we did, I rewrote this bit without using tls.